code/datums/craft/gun_parts.dm

```diff
@@ -53,6 +87,23 @@
 			barrel_attached = TRUE
 			to_chat(user, SPAN_NOTICE("You have attached the barrel to \the [src]."))
 			return
+
+	if(istool(I))
+		if(I.get_tool_quality(QUALITY_SCREW_DRIVING))
+			var/list/possibles = contents.Copy()
+			possibles += "Cancel"
+			var/obj/item/part/gun/toremove = input("Which part would you like to remove?","Removing parts") in possibles
+			if(toremove == "Cancel")
+				return
```

Comment:
```suggestion
			var/obj/item/part/gun/toremove = input("Which part would you like to remove?","Removing parts") in possibles
			if(!toremove)
				return
```
There is already a Cancel button to the input prompt, no need to add a "Cancel" option in the list.